asteroid/utils/hub_utils.py

```diff
@@ -39,11 +56,29 @@ def cached_download(filename_or_url):
     if os.path.isfile(filename_or_url):
         return filename_or_url
 
-    if filename_or_url in MODELS_URLS_HASHTABLE:
+    if urlparse(filename_or_url).scheme in ("http", "https"):
+        url = filename_or_url
+    elif filename_or_url in MODELS_URLS_HASHTABLE:
         url = MODELS_URLS_HASHTABLE[filename_or_url]
     else:
-        # Give a chance to direct URL, torch.hub will handle exceptions
-        url = filename_or_url
+        # Finally, let's try to find it on Hugging Face model hub
+        # e.g. julien-c/DPRNNTasNet-ks16_WHAM_sepclean is a valid model id
+        # and  julien-c/DPRNNTasNet-ks16_WHAM_sepclean@main supports specifying a commit/branch/tag.
+        if "@" in filename_or_url:
+            model_id = filename_or_url.split("@")[0]
+            revision = filename_or_url.split("@")[1]
+        else:
+            model_id = filename_or_url
+            revision = None
+        url = hf_bucket_url(model_id=model_id, filename=HF_WEIGHTS_NAME, revision=revision)
+        return hf_get_from_cache(url, cache_dir=get_cache_dir())
+        # Note to maintainers:
+        # You can remove the `return hf_get_from_cache(…)` line above
+        # if you want to keep the exact same file downloading/caching behavior
+        # as the current one. In which case, you can remove all functions below
+        # except for `hf_bucket_url`.
+        # However the implementation adds some nice features
+        # (notably versioning-aware caching) so I'd suggest keeping it.
```

Comment:
We'll keep `hf_get_from_cache` because we'll drop Zenodo support as soon as our models are fully migrated to HuggingFace's hub. 
And we'll add a note about this behaviour in the docstring above. 
Thanks for the note!